cookbooks/sidekiq6/recipes/setup.rb

```diff
@@ -0,0 +1,95 @@
+#
+# Cookbook Name:: sidekiq
+# Recipe:: setup
+#
+
+if node['sidekiq']['is_sidekiq_instance']
+  # report to dashboard
+  ey_cloud_report "sidekiq6" do
+    message "Setting up sidekiq 6 or later"
+  end
+
+  # loop through applications
+  node['dna']['applications'].each do |app_name, _|
+    node['sidekiq']['workers'].times do |count|
+      execute "restart-sidekiq-for-#{app_name}-#{count}" do
+        command "systemctl daemon-reload && systemctl restart sidekiq_#{app_name}_#{count}"
+        action :nothing
+      end
+
+    # set up systemd
+      template "/lib/systemd/system/sidekiq_#{app_name}_#{count}.service" do
+        mode 0644
+        source "sidekiq.service.erb"
+        backup false
+        variables({
+          :app_name => app_name,
+          :count => count,
+          :rails_env => node['dna']['environment']['framework_env'],
+          :memory_limit => node['sidekiq']['worker_memory']
+        })
+        notifies :run, "execute[restart-sidekiq-for-#{app_name}-#{count}]"
+    end
+end
+    execute "set-up-variables-for-sidekiq" do
+      command "cat /data/#{app_name}/shared/config/env.cloud |tr -d '\"' |awk '{gsub(\"export \", \"\");print}' > /data/#{app_name}/shared/config/env.sidekiq.cloud"
+    end
+    # database.yml
+    execute "update-database-yml-pg-pool-for-#{app_name}" do
+      db_yaml_file = "/data/#{app_name}/shared/config/database.yml"
+      command "sed -ibak --follow-symlinks 's/reconnect/pool:      #{node['sidekiq']['concurrency']}\\\n  reconnect/g' #{db_yaml_file}"
+      action :run
+      only_if "test -f #{db_yaml_file} && ! grep 'pool: *#{node['sidekiq']['concurrency']}' #{db_yaml_file}"
+      notifies :run, "execute[restart-sidekiq-for-#{app_name}]"
```

Comment:
As @herbertia pointed out, this doesn't work as-is – it's referring to a resource that doesn't exist. We changed this to:

```
      node['sidekiq']['workers'].times do |count|
        notifies :run, "execute[restart-sidekiq-for-#{app_name}-#{count}]"
      end
```